src/DualReader.php

```diff
@@ -54,14 +53,27 @@ public function getMethodAnnotations(ReflectionMethod $method): array
      */
     public function getClassAnnotations(ReflectionClass $class): array
     {
-        if ($this->php8) {
-            $annotations = $this->attributeReader->getClassAnnotations($class);
-            if ($annotations) {
-                return $annotations;
-            }
+        $annotations = $this->annotationReader->getClassAnnotations($class);
+        if (! $this->php8) {
+            return $annotations;
+        }
+
+        $attributes = $this->attributeReader->getClassAnnotations($class);
+        return array_merge($annotations, $attributes);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getPropertyAnnotations(ReflectionProperty $reflectionProperty): array
+    {
+        $annotations = $this->annotationReader->getPropertyAnnotations($reflectionProperty);
+        if (! $this->php8) {
+            return $annotations;
         }
 
-        return $this->delegate->getClassAnnotations($class);
+        $attributes = $this->attributeReader->getPropertyAnnotations($reflectionProperty);
+        return array_merge($annotations, $attributes);
     }
```

Comment:
This only moves `getPropertyAnnotations()` a bit higher, so all methods that return array `getXS()` all together.